File: algo/implementation/cpp/beautifulDaysAtMovie.cpp
#include <iostream>
#include <cstdlib>
using namespace std;

int rev(int n) {
    int reversedNumber = 0;
    while (n) {
        int remainder = n % 10;
        reversedNumber = reversedNumber * 10 + remainder;
        n = n / 10;
    }
    return reversedNumber;
}

int main() {
    int i, j, k;
    cin >> i >> j >> k;
    int count = 0;
    for (int start = i; start <= j; start++) {
        int reverse = rev(start);
        if (abs(reverse - start) % k == 0) {
            count += 1;
        }
    }
    cout << count;
    return 0;
}

File: algo/implementation/js/beautifulDaysAtMovie.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));
    main();
});

function readLine() {
    return inputString[currentLine++];
}

const rev = (n) => {
    let reversed = 0;
    while (n) {
        let remainder = n % 10;
        reversed = reversed * 10 + remainder;
        n = parseInt(n / 10);
    }
    return reversed;
}

function beautifulDays(i, j, k) {
    let count = 0;
    for (let start = i; start <= j; start += 1) {
        let reversed = rev(start);
        if (Math.abs(start - reversed) % k === 0) { count += 1; }
    }
    return count;

}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);
    const ijk = readLine().split(' ');
    const i = parseInt(ijk[0], 10);
    const j = parseInt(ijk[1], 10);
    const k = parseInt(ijk[2], 10);
    let result = beautifulDays(i, j, k);
    ws.write(result + "\n");
    ws.end();
}
